warehouse/legacy/api/json.py

```diff
@@ -155,12 +155,16 @@ def _json_data(request, project, release, *, all_releases):
             "requires_dist": (
                 list(release.requires_dist) if release.requires_dist else None
             ),
+            "provides_extra": (
+                list(release.provides_extra) if release.provides_extra else None
+            ),
```

Comment:
Should we make a distinction here between when the field is not provided, and when it is provided but empty? I'm not sure if the PEP addresses this or makes a distinction.

Reply:
Both `Provides-Extra` and `Dynamic` are optional multiple use fields, is at least `METADATA` files can't express a difference between missing-empty and present-empty, so i'd say it's fine to treat them the same here.

Reply:
I agree with @konstin - there's no meaningful way of interpreting "present but empty" *except* by not including the metadata item in the `METADATA` file, so the two are for all practical purposes equivalent.

Reply:
Since both results are functionally the same I think either approach is fine.